File: DP/Q1.cpp
class Solution {
public:
    int fib(int n) {
        int result[31]={0};
        int i=0;
        if(n==0 || n==1){
            return n;
        }
        if(result[n]==0){
            result[n]=fib(n-1)+fib(n-2);   
        }
        return result[n];
    }
};

File: PrefixSum/Q1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        l = len(nums)
        result = []
        a = 1
        for n in nums:
            result.append(a)
            a*=n
        a = 1
        for j in reversed(range(l)):
            result[j]*=a
            a*=nums[j]
        return result
